cqrs-postgres/src/db_wrapper.rs

```diff
@@ -0,0 +1,177 @@
+use crate::util::Sequence;
+use cqrs_core::{CqrsError, EventNumber, RawEvent, Since};
+use num_traits::ToPrimitive;
+use postgres::{rows::Row, types::ToSql};
+use r2d2::{Pool, PooledConnection};
+use r2d2_postgres::PostgresConnectionManager;
+use std::{error, fmt, sync::Arc};
+
+#[derive(Debug, Clone)]
+pub enum DbError {
+    Format(Arc<dyn CqrsError>),
+    Pool(Arc<dyn CqrsError>),
+    Postgres(Arc<dyn CqrsError>),
+    React(Arc<dyn CqrsError>),
```

Comment:
See if you can make this `React(Arc<R::Error>)` with the generic proposed above. You may even be able to get away with `React(R::Error)` directly.